File: Data Structures & Algorithms/sort-colors/submission-2.java
class Solution {
    public void sortColors(int[] nums) {
        int start = 0, end = nums.length - 1, mid = 0;
        while(mid <= end){
            if(nums[mid] == 0){
                swap(nums, start, mid);
                mid++;
                start++;
            }
            else if(nums[mid] == 1){
                mid++;
            }
            else{
                swap(nums, end, mid);
                end--;
            }
        }
    }

    public void swap(int[] nums, int a, int b){
        int temp = nums[a];
        nums[a] = nums[b];
        nums[b] = temp;
    }
}

File: Data Structures & Algorithms/search-2d-matrix/submission-2.py
class Solution:
    def searchMatrix(self, matrix: List[List[int]], target: int) -> bool:
        row = len(matrix)
        col = len(matrix[0])
        
        start, end = 0, row*col - 1

        while start <= end:
            mid = (start+end)//2
            midval = matrix[mid//col][mid%col]

            if midval == target:
                return True
            elif midval > target:
                end = mid - 1
            else:
                start = mid + 1
        return False